### ISSUE TITLE
Consumed messages with small size have trailing garbage.

### ISSUE THREAD

**Issue by ZLyle** [CopernicaMarketingSoftware/AMQP-CPP#252]
```#include <string>

#include <boost/asio/io_service.hpp>
#include <boost/asio/strand.hpp>

#include <amqpcpp.h>
#include <amqpcpp/linux_tcp.h>
#include <amqpcpp/libboostasio.h>

int main() {
  boost::asio::io_service service(2);

  AMQP::LibBoostAsioHandler handler(service);

  AMQP::TcpConnection connection(
      &handler, AMQP::Address("amqp://guest:guest@localhost:7777//"));

  AMQP::TcpChannel channel(&connection);

  const std::string queue_name{"cpp_test_queue"};

  channel.declareQueue(queue_name, AMQP::durable)
      .onSuccess([&channel, &queue_name](const std::string& name,
                                         uint32_t           message_count,
                                         uint32_t           consumer_count) {
        std::cout << "declared queue " << name << ", has " << message_count
                  << " messages and " << consumer_count << " consumers."
                  << std::endl;
        channel.publish("", queue_name, "testing 1 2 3... etc etc");
        channel.publish("", queue_name, "testing 4 5 6... etc");
        channel.publish("", queue_name, "testing 7 8 9...");
      });

  auto message_callback = [&channel](const AMQP::Message& message,
                                     uint64_t             delivery_tag,
                                     bool                 redelivered) {
    std::string msg{message.body()};
    //msg.resize(message.bodySize());
    std::cout << "Received: " << msg << " with tag " << delivery_tag
              << " and it " << (redelivered ? "was" : "wasn't")
              << " redelivered." << std::endl;
    channel.ack(delivery_tag);
  };

  channel.consume(queue_name).onReceived(message_callback);

  service.run();

  connection.close();
```

Produces the output:

```declared queue cpp_test_queue, has 0 messages and 0 consumers.
Received: testing 1 2 3... etc etcÎ6qC5a6Vombw with tag 1 and it wasn't redelivered.
Received: testing 4 5 6... etcÎ2vk46qC5a6Vombw with tag 2 and it wasn't redelivered.
Received: testing 7 8 9...ÎkjdU2vk46qC5a6Vombw with tag 3 and it wasn't redelivered.
```

Where the garbage is always random and always results in a string of that length. I've tested with libev as well as boostasio, the results are the same. Examining the messages with pika in python shows no garbage, and messages from previous publishes (without consuming them immediately as I do in my example) have the same garbage, so it probably isn't an issue with publishing or what I'm publishing. bodySize() reports the correct size.

Also, messages of any size always end with Î for me, which is might be a different issue entirely.

**Comment by lazedo** [comment]
try `std::string msg(message.body(), message.bodySize());`
body is const char * but not null terminated

**Comment by ZLyle** [comment]
That makes sense, I was unaware of this fact but kind of suspected it. At one point I added put a literal `\0` in the string I published, but it still printed garbage so I made the assumption the messages were null terminated and something else was the issue.

Your solution works as expected, and it's good to know about the lack of null termination. I'm going to close this under the assumption that this is simply my own mistake.
